posthog/temporal/data_imports/pipelines/pipeline.py

```diff
@@ -70,13 +94,28 @@ def _create_pipeline(self):
 
     def _run(self) -> Dict[str, int]:
         pipeline = self._create_pipeline()
-        pipeline.run(self.source, loader_file_format=self.loader_file_format)
 
-        row_counts = pipeline.last_trace.last_normalize_info.row_counts
-        # Remove any DLT tables from the counts
-        filtered_rows = filter(lambda pair: not pair[0].startswith("_dlt"), row_counts.items())
+        total_counts: Counter = Counter({})
+
+        if self._incremental:
+            counts: Counter = Counter({})
+
+            while counts:
+                pipeline.run(self.source, loader_file_format=self.loader_file_format)
+
+                row_counts = pipeline.last_trace.last_normalize_info.row_counts
+                # Remove any DLT tables from the counts
+                filtered_rows = filter(lambda pair: not pair[0].startswith("_dlt"), row_counts.items())
+                counts = Counter(dict(filtered_rows))
+                total_counts = counts + total_counts
```

Comment:
So, does this run the pipeline repeatedly in 500-row increments until the `counts` returns `0`?

Reply:
yep